hash_practice/exercises.py

```diff
@@ -22,8 +48,38 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2)
+        Space Complexity: O(n)

```

Comment:
These calculations are correct given your current algorithm, but a full solution will probably push the time complexity up.

At the same time, since we assume all grids are 3x3 the number of steps and space used will always be constant so you could consider this to be O(1) in space and time complexity. Of course,  allowing for any size of boards will change that, but in this case, it's always 3x3.